### PR TITLE
[SPARK-49177][INFRA][DOCS] Make GA's `build_error_docs` run only once

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to make GA's `build_error_docs` run only `once`.

### Why are the changes needed?
- At present, from observing the logs of `GA`, we have found that `build_error_docs` has run `twice`, which is `unnecessary`, as follows:
  image

- Additionally, in step `Run documentation build` of `GA`, some environment variables (eg: `SKIP_PYTHONDOC`, `SKIP_RDOC`) are dynamic. I propose printing their values before executing `bundle exec jekyll build` to facilitate developer debugging.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manually check.
- Before
  image

- After
  image

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @HyukjinKwon @dongjoon-hyun

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Sorry, but this looks like not a good approach, @panbingkun , because `build_error_docs` is `non-API` docs which is outside of `https://spark.apache.org/docs/latest/api` path.

https://spark.apache.org/docs/latest/sql-error-conditions.html

If this is important, you may add `SKIP_ERRORDOC` in a similar way. WDYT?

**Comment by panbingkun** [comment]
> SKIP_ERRORDOC

Makes sense, I have already updated it, thanks!

File: docs/_plugins/build_api_docs.rb [hunk 1]
```diff
@@ -203,7 +203,9 @@ def build_error_docs
   || raise("Error doc generation failed")
 end
 
-build_error_docs
+if not (ENV['SKIP_ERRORDOC'] == '1')
```
Comment (dongjoon-hyun) [inline docs/_plugins/build_api_docs.rb:206]: Thank you!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM from my side.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**